packages/browser/src/core/task/task-group.ts

```diff
@@ -0,0 +1,22 @@
+export class TaskGroup {
+  private promise!: Promise<void>
+  private res!: () => void
+  count = 0
+
+  done = () => this.promise
+  run = <Operation extends (...args: any[]) => any>(
```

Comment:
Another way to type this is “<ReturnT, Operation extends (…args: unknown[]) => T>(op: Operation): ReturnT